File: Programação e Sistemas de Informação/M9 - Introdução à Programação Orientada a Objetos/Ex6/Circulo.java

//
// Copyright (c) Carlos Tojal (24-10-2019)
// Ex6
// Circulo.java
//

public class Circulo {
    double x, y, raio;

    public double getRaio() {
        return raio;
    }

    public Circulo() {
    }

    public Circulo(double x, double y, double raio) {
        this.x = x;
        this.x = x;
        this.raio = raio;
    }

    public double calcularPerimetro() {
        return 2 * Math.PI * raio;
    }

    public double calcularArea() {
        return Math.PI * Math.pow(raio, 2);
    }

    public void aumentarRaio(double num) {
        raio += num;
    }

    public boolean verificarRaio(Circulo circulo) {
        return this.raio > circulo.getRaio(); 
    }
}

File: Programação e Sistemas de Informação/M9 - Introdução à Programação Orientada a Objetos/Ex6Kotlin/Circulo.kt

//
// Copyright (c) Carlos Tojal (24-10-2019)
// Ex6Kotlin
// Circulo.kt
//

import kotlin.math;

class Circulo(x: Double, y: Double, raio: Double, raio_aumentado: Double) {
    
    constructor(): this(0.0, 0.0, 0.0, 0.0);
    constructor(x: Double, y: Double, raio: Double): this(x, y, raio, 0.0);

}